### DISCUSSION TITLE
Kokkos - discards qualifiers - const

### DISCUSSION THREAD

**Discussion by ARSanderson** [arborx/ArborX#686] [General]
In a code that I am compiling ArborX is used. I am getting a couple of errors that while I completely understand the issue I am not sure how to proceed. Below are the basics. For the machine being used it was just upgraded to RockyLinux8 and we use a cuda@11.5.0 and cuda_arch of 70 for the nodes when building with spack.  As the machine was just updated and well as updating the calling code base I am not sure if ArborX is new in the project or not (checking on that now).

Any guidance would be appreciated.

236  /uufs/chpc.utah.edu/common/home/u0059283/Plato_new/spack/opt/spack/linux-rocky8-x86_64/gcc-8.5.0/arborx-header_only-p4pqzc6zmcik5owrlkremkc53yz7dd25/include/details/ArborX_Detail
            sBoundingVolumeHierarchyImpl.hpp:423:57: error: binding reference of type 'Kokkos::View<int*, Kokkos::Device<Kokkos::Cuda, Kokkos::CudaSpace> >&' to 'const Kokkos::View<int*, Kok
            kos::Device<Kokkos::Cuda, Kokkos::CudaSpace> >' discards qualifiers
     237               Details::TreeTraversal<DeviceType>::query(
     238                                                             ^
 239    /uufs/chpc.utah.edu/common/home/u0059283/Plato_new/spack/opt/spack/linux-rocky8-x86_64/gcc-8.5.0/arborx-header_only-p4pqzc6zmcik5owrlkremkc53yz7dd25/include/details/ArborX_Detail
            sBoundingVolumeHierarchyImpl.hpp:515:57: error: binding reference of type 'Kokkos::View<int*, Kokkos::Device<Kokkos::Cuda, Kokkos::CudaSpace> >&' to 'const Kokkos::View<int*, Kok
            kos::Device<Kokkos::Cuda, Kokkos::CudaSpace> >' discards qualifiers
     240               Details::TreeTraversal<DeviceType>::query(

**Comment by dalg24** [comment]
It's hard to tell much if you don't say what version of ArborX you are using.
Most likely the issue is in the code that calls ArborX.
My best guess is that code is passing `const`-qualified views `indices` and `offsets` (or `values` and `offsets`) to the `query` function.
Also, the error log you posted is truncated.

**Comment by ARSanderson** [comment]
Thank you for the quick reply. From the spack spec it is the header only version of ArborX which they have patched only the CMakeLists.txt file.

 ^arborx@header_only%gcc@8.5.0~cuda~ipo~mpi~openmp~serial build_type=RelWithDebInfo patches=89837fe3c7920c61871e8955cdf00788caab55227738b6b70decdf8baafcf706 arch=linux-rocky8-skylake_avx512

I am pinging my colleagues who wrote the base code. I have attached the full error log as a zip file as it is quite verbose. If there are any other clues please let me know.  Especially, as the machine being recently updated I want make there is not a incompatibility. 

[Plato_issue.tex.zip](https://github.com/arborx/ArborX/files/8892848/Plato_issue.tex.zip)

**Comment by aprokop** [comment]
This is a pretty old version of the code (pre 1.0). `ArborX_DetailsBoundingVolumeHieararchyImpl.hpp` was removed in #425 about 18 months ago.

**Reply by ARSanderson** [reply]
Thanks, I will let my colleagues know and go from there. I greatly appreciate the quick replies. Thank you

**Reply by dalg24** [reply]
Would also please bring platoengine/platoanalyze#2, platoengine/platoanalyze#3, and platoengine/platoanalyze#101 to their attention?

**Comment by ARSanderson** [comment]
Done, thanks.
